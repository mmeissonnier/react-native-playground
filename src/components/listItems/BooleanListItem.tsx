import React, { useState, StatelessComponent } from 'react';
import { View, Text, Switch } from 'react-native';
import { ListItemProps } from '../../types';

const BooleanListItem: StatelessComponent<ListItemProps<boolean>> = ({
  id,
  label,
  value = false,
  onChange,
}) => {
  const [state, setState] = useState({ checked: value });

  return (
    <View
      style={{
        display: 'flex',
        flexDirection: 'row',
        width: '100%',
        alignItems: 'center',
      }}
    >
      <Text style={{ fontSize: 20, fontWeight: '200', color: 'white' }}>
        {label}
      </Text>

      <Switch
        value={state.checked}
        onValueChange={newVal => {
          if (onChange) {
            onChange(id, newVal);
          }
          setState({ checked: newVal });
        }}
      />
    </View>
  );
};

export default React.memo(BooleanListItem);
