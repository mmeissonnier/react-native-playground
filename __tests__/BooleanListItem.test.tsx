import React from 'react';
import { shallow } from 'enzyme';
import BooleanListItem from '../src/components/listItems/BooleanListItem';

test('It should render correctly', () => {
  expect(
    shallow(<BooleanListItem id="1" label="Bar" value={true} />)
  ).toMatchSnapshot();
});
