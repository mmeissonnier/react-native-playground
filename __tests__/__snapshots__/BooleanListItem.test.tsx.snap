// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "200",
      }
    }
  >
    Bar
  </Text>
  <Switch
    onValueChange={[Function]}
    value={true}
  />
</View>
`;
